roomba/main.c

```diff
@@ -0,0 +1,173 @@
+    }
+    return true;
+}
+
+void setuprobot(bool debug, bool camera){
+    create_connect();
+    setupsensordata();
+    if(debug == true){
+        printsensordata();
+    }
+    if(camera){
+        if(setupcamera()){
+            printf("%s","Camera initialized");
+        }
+        else{
+            printf("%s","Camera Error");
+        }
+    }
+    enable_servos();
+}
+
+void shutdownrobot(bool camera){
+    if(camera){
+        camera_close();
+    }
+    disable_servos();
+    ao();
+    create_stop();
+    create_disconnect();
+}
+
+void drivestraight(int speed, bool direction, int distance, bool watchfortouch){ //false backwards true forwards distance in mm
+    if(direction == true){
+        set_create_distance(0);
+        while(get_create_distance() <= (distance)){
+            create_drive_direct(speed,speed);
+            if(watchfortouch){
+                if(get_create_lbump() || get_create_rbump() == true){
+                    set_create_distance(distance);
+                }
+            }
+        }
+    }
+    else{
+        set_create_distance(0);
+        while((-1*get_create_distance()) <= (distance)){
+            create_drive_direct(speed*-1,speed*-1);
+        }
+    }
+}
+void turn(int speed, int deg){
+    // deg - nach rechts + nach links
+    set_create_distance(0);
+    set_create_total_angle(0);
+    if(deg>0){
+     while(get_create_total_angle() <= deg){
+         create_drive_direct(speed * -1,speed);
+     }
+    }
+    else{
+            while(get_create_total_angle() >= deg){
+                create_drive_direct(speed,speed * - 1);
+            }
+        }
+}
+
+void wasd(){
+    bool exit;
+    exit = true;
+    int speed = 100;
+    while(exit){
+        char input;
+        printf("%s","Please provide input:\n");
+        scanf("%c",&input);
+        if(input == 'w'){
+            drivestraight(speed,true,25,true);
+        }
+        if(input == 's'){
+            drivestraight(speed,false,25,true);
+        }
+        if(input == 'd'){
+            turn(speed/2,-45);
+        }
+        if(input == 'a'){
+            turn(speed/2,45);
+        }
+        if(input == 'x'){
+            exit = false;
+        }
+        if(input == 'm'){
+            printf("%s","Please provide speed:\n");
+            scanf("%d",&speed);
+        }
+    }
+}
+
+void grapscher(int startposition, int endposition, int port){ // opens and closes claw
+    if(get_servo_position(port) < startposition){
+        moveservo(port,startposition);
+    }
+    moveservo(port,endposition);
+}
+
+void movearm(int position, bool grabobject,int rotation, int lowerservoid, int grabservoid, int speed){
+    int idlegrabposition = 500; // placeholder
+    if(rotation != 0){ // turn
+        turn(speed, rotation);
+    }
+    moveservo(lowerservoid,position);
+    // claw
+    if(grabobject == true){
+        grapscher(idlegrabposition,550,grabservoid); //placeholder value
+    }
+}
+
+int getObject(int channel){
```

Comment:
This function does not return an ``integer``!

Opinionated: You can reduce unnecessary nesting by inverting the ``if-statement``:

```c
void getObject(int channel){
    if(!get_object_count(channel)) return;
    camera_update();
    
    get_object_center_x(channel, 0);
    get_object_center_y(channel, 0);
}
```